native/src/main/java/io/ballerina/wso2/apim/catalog/ServiceCatalog.java

```diff
@@ -0,0 +1,265 @@
+    }
+
+    private static boolean isHttpServiceNode(Object listenerDetails) {
+        return (((ArrayList<BObject>) listenerDetails).get(0)).getOriginalType().
+                getPackage().getName().equals(HTTP_ORG_NAME);
+    }
+
+    private static Object getAnnotations(Object serviceObj) {
+        if (serviceObj == null) {
+            return null;
+        }
+        BObject serviceBObj = (BObject) serviceObj;
+        ObjectType impliedType = (ObjectType) TypeUtils.getImpliedType(serviceBObj.getOriginalType());
+        return impliedType.getAnnotations();
+    }
+
+    private static void updateMetadata(Environment env, BMap<BString, Object> artifactValues, Object listenerDetails,
+                                     Object annotationDetails, Object attachPointDetails) {
+        BMap<BString, Object> httpAnnotation = getHttpAnnotation((BMap<BString, Object>) annotationDetails);
+        HttpServiceConfig httpServiceConfig = updateHostAndPortAndBasePath(listenerDetails,
+                attachPointDetails, httpAnnotation);
+        updateServiceNameAndUrl(artifactValues, httpServiceConfig);
+        updateAnnotationsArtifactValues(artifactValues, annotationDetails, httpAnnotation, env, httpServiceConfig);
+        updateListenerConfigurations(artifactValues, listenerDetails);
+        updateMd5(artifactValues);
+    }
+
+    private static void updateMd5(BMap<BString, Object> artifactValues) {
+        String name = StringUtils.getStringValue(artifactValues.get(StringUtils.fromString(NAME)));
+        String version = StringUtils.getStringValue(artifactValues.get(StringUtils.fromString(VERSION)));
+        String definitionType = StringUtils.getStringValue(artifactValues.get(StringUtils.fromString(DEFINITION_TYPE)));
+        String openapiDef = StringUtils.getStringValue(artifactValues.get(StringUtils
+                .fromString(DEFINITION_FILE_CONTENT)));
+
+        String string = new StringBuilder(name).append(version).append(definitionType).append(openapiDef).toString();
+        artifactValues.put(StringUtils.fromString(MD5), StringUtils.fromString(createMd5Hash(string)));
+    }
+
+
+    private static void updateServiceNameAndUrl(BMap<BString, Object> artifactValues,
+                                                HttpServiceConfig httpServiceConfig) {
+        updateServiceName(artifactValues, httpServiceConfig);
+        updateServiceUrl(artifactValues, httpServiceConfig);
+    }
+
+    private static void updateListenerConfigurations(BMap<BString, Object> artifactValues, Object listenerDetails) {
+        artifactValues.put(StringUtils.fromString(MUTUAL_SSL_ENABLED), getMutualSSLDetails(listenerDetails));
+    }
+
+
+    private static boolean getMutualSSLDetails(Object listenerDetails) {
+        ArrayList<BObject> listenerArray = (ArrayList<BObject>) listenerDetails;
+        if (listenerArray.size() > 0) {
+            BObject listener = listenerArray.get(0);
+            if (listener != null) {
```

Comment:
Do we add null to listnerArray? otherwise I think this should always be false since we have already verified list is not empty

Reply:
We dont add it in this module, But there can be a possibility that listener is null